### DISCUSSION TITLE
How can I add an element in Stencil to paper by clicking on the element instead of dragging and dropping it?

### DISCUSSION THREAD

**Discussion by tkaabbc** [clientIO/joint#2472] [Q&A] [answered]
### Introduction

Hi Team,

I want to add elements in Stencil to paper **by clicking on them instead of dragging and dropping**

but I didn't find a config or an event that I can listen to the stencil element clicking. If you could give me any tips, I'd be grateful.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment, marked as answer]
The stencil is a set of papers (`dia.Paper`). You can retrieve any paper with `stencil.getPaper()` method and bind any events to it.
```js
stencil.options.canDrag = () => false;
stencil.getPaper('my-stencil-group').on('element:pointerclick', function(elementView) {
    const clone = elementView.model.clone();
    clone.position(100, 100);
    graph.addCell(clone);
});
```

**Reply by tkaabbc** [reply]
Thanks a lot, it works! 
By the way, I need to bind the event to all stencil groups, so finally, I implement it like the below
```js
Object.values(stencil.papers).forEach((paper) => {
      paper.on('element:pointerclick', (elementView) => {
        const clone = elementView.model.clone();
        clone.position(100, 100);
        this.graph.addCell(clone);
      });
    });
  ```
